src/tcp.cpp

```diff
@@ -120,9 +121,21 @@ TCP::TCP(const uint8_t* buffer, uint32_t total_sz) {
             stream.skip(len);
         }
     }
+    
     // If we still have any bytes left
     if (stream) {
-        inner_pdu(new RawPDU(stream.pointer(), stream.size()));
+        PDU* new_pdu = nullptr;
+        if ((new_pdu = Internals::allocate<TCP>(
+                 {Allocators::SRC_PORT, sport()}, stream.pointer(), stream.size()))) {
+        }
+        else if ((new_pdu = Internals::allocate<TCP>(
+                      {Allocators::DST_PORT, dport()}, stream.pointer(), stream.size()))) {
```

Comment:
Nit: alignment should match line 129 (also in udp.cpp)

Reply:
I agree that those should line up, but this is how emacs indents it -- based on the open parenthesis on the previous line, maybe? I created a config specifically for this library to match the upstream author's style. It's been good so far, so I'd rather not adjust this line.